tests/integration-tests/conftest.py

```diff
@@ -1394,12 +1397,23 @@ def _run_benchmarks(remote_command_executor, scheduler_commands, **kwargs):
                         test_datadir,
                         dimensions,
                     )
+                    # Load reference results to confront benchmark and Assert failure
+                    if check_result:
+                        reference_results = results_csv_to_dict()
+                        check_errors = []
+
                     for metric_data in metric_data_list:
+                        # Check threshold
+                        if check_result:
+                            check_errors = check_threshold(reference_results, metric_data)
+
                         cloudwatch_client.put_metric_data(
                             Namespace=metric_namespace,
                             MetricData=metric_data,
                         )
         logging.info("Finished benchmarks for %s", function_name)
+        if check_result:
```

Comment:
let's remove this if and always return check_errors even if it's an empty list and let's add in the docstring of the function what we're returning. In this way the caller can have a consistent workflow in both the cases.